adiv5: use PRIxNN specifiers for debug printfs

xobs opened the pull request:
This fixes a few printf-style warnings that slipped through due to the
fact that they're only enabled on debug builds. This enables building
this file on an Xtensa platform where -Werror and -Wformat is enabled by
default.

dragonmux commented on src/target/adiv5.c:
```diff
@@ -721,7 +721,7 @@ void adiv5_dp_init(ADIv5_DP_t *dp)
 		/* Check for a valid DPIDR / designer */
 		if (dp->designer_code != 0) {
-			DEBUG_INFO("DP DPIDR 0x%08" PRIx32 " (v%d %srev%d) designer 0x%" PRIx32 " partno 0x%" PRIx32 "\n", dpidr,
+			DEBUG_INFO("DP DPIDR 0x%08" PRIx32 " (v%"PRIx8" %srev%"PRId32") designer 0x%" PRIx16 " partno 0x%" PRIx16 "\n", dpidr,
```
PRIx16 and PRIx8 doesn't work on nano newlibc last we tried. These should use %x

xobs replied:
I've switched it to `%x`, which still compiles on ESP32.

dragonmux commented on src/target/adiv5.c:
```diff
@@ -753,7 +753,7 @@ void adiv5_dp_init(ADIv5_DP_t *dp)
 		dp->target_partno = (targetid & ADIV5_DP_TARGETID_TPARTNO_MASK) >> ADIV5_DP_TARGETID_TPARTNO_OFFSET;
 
-		DEBUG_INFO("TARGETID 0x%08" PRIx32 " designer 0x%" PRIx32 " partno 0x%" PRIx32 "\n", targetid,
+		DEBUG_INFO("TARGETID 0x%08" PRIx32 " designer 0x%" PRIx16 " partno 0x%" PRIx16 "\n", targetid,
```
This duplicates #1163 but suffers from the same problem as the above format string.

xobs replied:
I've also switched this to `%x`, and confirmed that it still builds.

dragonmux reviewed (changes requested):
Thank you for fixing some extra instances of format specifiers being wrong! Much appreciated. We're in dire need for Esden reviewing and merging our PRs right now as this PR partially duplicates one of ours (not your fault) - if you wish to fully consume the changes we made in that PR, we'll close it in favour of this one.

xobs commented:
I've updated changes and force-pushed the branch. I've verified that the changes still build on ESP32. Additionally, I've added the same fix to `adiv5_jtagdp.c`.

dragonmux reviewed (approved):
LGTM, considering this as replaces #1163 and merging in favour of that PR.
